File: aoj/volume100/java/A10026.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class A10026 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        while (true) {
            int n = scan.nextInt();
            if (n == 0)
                break;

            List<Integer> scores = new ArrayList<Integer>();
            for (int i = 0; i < n; i++)
                scores.add(scan.nextInt());

            double average = 0;
            for (int score : scores) {
                average += score;
            }
            average /= scores.size();

            double variance = 0.0;
            for (int score : scores) {
                variance += Math.pow(score - average, 2);
            }
            variance /= scores.size();

            double standardDeviation = Math.sqrt(variance);
            System.out.println(standardDeviation);
        }
    }
}

File: aoj/volume100/ruby/A10026.rb
while line = gets do
  n_items = line.to_i
  break if n_items == 0

  scores   = gets.split.map(&:to_f)
  average  = scores.inject(:+).to_f / scores.size
  variance = scores.inject(0.0){|sum, score| sum += (score-average)**2} / scores.size

  standard_deviation = variance ** 0.5
  puts standard_deviation.round(6)
end
